src/devtools/ProfilingCache.js

```diff
@@ -225,13 +243,14 @@ export default class ProfilingCache {
       return new Promise(resolve => {
         const importedProfilingData = this._store.importedProfilingData;
         if (importedProfilingData !== null) {
-          const { profilingSummary } = (importedProfilingData: any);
-          if (profilingSummary != null) {
-            this._pendingProfileSummaryMap.set(
-              profilingSummary.rootID,
-              resolve
-            );
-            this.onProfileSummary(profilingSummary);
+          const profilingSummaryFrontend: ProfilingSummaryFrontend =
+            importedProfilingData.profilingSummary;
+          if (
+            profilingSummaryFrontend != null &&
+            profilingSummaryFrontend.rootID === rootID
```

Comment:
`selectedCommitIndex` is `null` here:
```
function CommitFlamegraphAutoSizer(_) {
  const _useContext = Object(react__WEBPACK_IMPORTED_MODULE_0__["useContext"])(_context__WEBPACK_IMPORTED_MODULE_8__["StoreContext"]),
        profilingCache = _useContext.profilingCache;

  const _useContext2 = Object(react__WEBPACK_IMPORTED_MODULE_0__["useContext"])(_ProfilerContext__WEBPACK_IMPORTED_MODULE_3__["ProfilerContext"]),
        rendererID = _useContext2.rendererID,
        rootID = _useContext2.rootID,
        selectedCommitIndex = _useContext2.selectedCommitIndex,
        selectFiber = _useContext2.selectFiber;

  const deselectCurrentFiber = Object(react__WEBPACK_IMPORTED_MODULE_0__["useCallback"])(event => {
    event.stopPropagation();
    selectFiber(null, null);
  }, [selectFiber]);
  const profilingSummary = profilingCache.ProfilingSummary.read({
    rendererID: rendererID,
    rootID: rootID
  });
  let commitDetails = null;
  let commitTree = null;
  let chartData = null;

  if (selectedCommitIndex !== null) {
    commitDetails = profilingCache.CommitDetails.read({
      commitIndex: selectedCommitIndex,
      rendererID: rendererID,
      rootID: rootID
    });
    commitTree = profilingCache.getCommitTree({
      commitIndex: selectedCommitIndex,
      profilingSummary
    });
    chartData = profilingCache.getFlamegraphChartData({
      commitDetails,
      commitIndex: selectedCommitIndex,
      commitTree
    });
  }

  if (commitDetails != null && commitTree != null && chartData != null && chartData.depth > 0) {
```

Looks like `ProfilingSummary` cache is caching forever for `rendererID === null` and `rootID === null` (`const key = hashInput(input);` is `null`) which are the values for when a file is imported.

```
function SnapshotSelector(_) {
  const _useContext = Object(react__WEBPACK_IMPORTED_MODULE_0__["useContext"])(_ProfilerContext__WEBPACK_IMPORTED_MODULE_3__["ProfilerContext"]),
        isCommitFilterEnabled = _useContext.isCommitFilterEnabled,
        minCommitDuration = _useContext.minCommitDuration,
        rendererID = _useContext.rendererID,
        rootID = _useContext.rootID,
        selectedCommitIndex = _useContext.selectedCommitIndex,
        selectCommitIndex = _useContext.selectCommitIndex;

  const _useContext2 = Object(react__WEBPACK_IMPORTED_MODULE_0__["useContext"])(_context__WEBPACK_IMPORTED_MODULE_6__["StoreContext"]),
        profilingCache = _useContext2.profilingCache;

  const _profilingCache$Profi = profilingCache.ProfilingSummary.read({
    rendererID: rendererID,
    rootID: rootID
  }),
```

And this `profilingSummaryFrontend.rootID === rootID` check is always false, so it resolves to default empty values:
```
    _defineProperty(this, "ProfilingSummary", Object(_cache__WEBPACK_IMPORTED_MODULE_0__["createResource"])(({
      rendererID,
      rootID
    }) => {
      return new Promise(resolve => {
        const importedProfilingData = this._store.importedProfilingData;

        if (importedProfilingData !== null) {
          const profilingSummaryFrontend = importedProfilingData.profilingSummary;

          if (profilingSummaryFrontend != null && profilingSummaryFrontend.rootID === rootID) {
            this._pendingProfileSummaryMap.delete(rootID);

            resolve(profilingSummaryFrontend);
            return;
```

When I added this check, I assumed that `rootID` is present when we import data, but it may be `null`, whereas `profilingSummaryFrontend.rootID` is always present.

Also we probably cannot use `profilingSummaryFrontend.rootID` from the imported data in the `resolve` as it theoretically has nothing to do with the currently loaded app's `rootID`s (if we load the profile while sitting on another app's page). But maybe we assume that if you load a profile, then it's from the same app that you're sitting at. Anyway, that `rootID` doesn't match the currently selected `rootID` so most likely useless and even confusing downstream.

Reply:
After I removed this check, the DevTools started showing an error `An error was thrown: "Could not find root node with id "null" in commit tree"`.

Reply:
OK fixed, missed one more check.

Also there was this type mismatch (`resolve` accepts `FiberCommitsBackend` but was given `FiberCommitsFrontend`, and thanks to Flow's non-nominal typing, this was ignored because they are structurally compatible):
```diff
-          this.onFiberCommits(fiberCommitsFrontend);
+          this._pendingFiberCommitsMap.delete(pendingKey);
+          resolve(fiberCommitsFrontend);
```

There were also missing cache invalidations:
```diff
      this._pendingCommitDetailsMap.clear();
+     this._pendingFiberCommitsMap.clear();
+     this._pendingInteractionsMap.clear();
      this._pendingProfileSummaryMap.clear();
```